### PR TITLE
Watchlist

### PR DESCRIPTION
**Pull request by Dklionsk**
One thing the app was missing was an example of a button. Buttons and the target-action pattern are useful things to teach, so I added an "Add to Watchlist" button to at the bottom of the MovieDetailVC. Clicking this button adds or removes the current movie from the user's watchlist. The watchlist is managed by an instance of WatchlistClient.

My comments:
* The watchlist doesn't need to be super fancy so I'm just persisting it to NSUserDefaults.
* Design-wise, if this were a real app, I would have put the add button in the navigation bar, but I wanted to teach how to add and configure buttons in IB.
* I switched the private client properties to being non-optional let properties and define them in custom initializers. I think this is a better pattern than keeping them as optionals.
* I'm doing if let movie = movie in a few places now. It's safe, but might be the wrong pattern for this.

### CODE REVIEW

File: Posterboard/MovieDetailVC.swift [hunk 1]
```diff
@@ -13,8 +13,20 @@ class MovieDetailVC: UIViewController {
...
     
-    var movieClient: MovieClient?
+    private let movieClient: MovieClient
+    private let watchlistClient: WatchlistClient
+    
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    required init(nibName: String?, bundle: NSBundle?, movieClient: MovieClient, watchlistClient: WatchlistClient) {
```
Comment (dennda) [inline Posterboard/MovieDetailVC.swift]: Any reason you pass nibName and bundle through here? I think you could just call super with the appropriate values and make the initializer shorter.

File: Posterboard/MovieDetailVC.swift [hunk 2]
```diff
@@ -53,10 +65,31 @@ class MovieDetailVC: UIViewController {
...
+            addButton.setTitle("Remove from Watchlist", forState: .Normal)
+        }
+        else {
+            addButton.setTitle("Add to Watchlist", forState: .Normal)
+        }
+    }
+    
+    @IBAction func clickedAddToWatchlist(sender: UIButton) {
+        if watchlistClient.movieInWatchlist(movie!.id) {
+            watchlistClient.removeFromWatchlist(movie!.id)
```
Comment (dennda) [inline Posterboard/MovieDetailVC.swift]: Not a big fan of the force unwraps here.

File: Posterboard/MovieDetailVC.swift (hunk 1 above)
Comment (Dklionsk) [inline Posterboard/MovieDetailVC.swift]: Good call

File: Posterboard/MovieDetailVC.swift (hunk 2 above)
Comment (Dklionsk) [inline Posterboard/MovieDetailVC.swift]: I fixed this

**Pull request merged**